opal/mca/accelerator/accelerator.h

```diff
@@ -0,0 +1,295 @@
+ */
+typedef int (*opal_accelerator_base_module_get_address_range_fn_t)(
+    const void *ptr, void **base, size_t *size);
+
+/**
+ * Queries whether the device supports Inter Process Communication
+ * or not. If true, the functions:
+ *
+ * opal_accelerator_base_module_get_handle_fn_t()
+ * opal_accelerator_base_module_open_handle_fn_t()
+ * opal_accelerator_base_module_close_handle_fn_t()
+ *
+ * must be implemented.
+ *
+ * @return true              IPC supported
+ * @return false             IPC not supported
+ */
+typedef bool (*opal_accelerator_base_module_is_ipc_enabled_fn_t)(void);
+
+
+/**
+ * Gets an interprocess memory handle for an existing device memory allocation.
+ *
+ * @param[IN] dev_ptr        Base pointer to previously allocated device
+ *                           memory
+ * @param[OUT] handle        Pointer to user allocated mem handle to return
+ *                           the memory handle in.
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ *
+ */
+typedef int (*opal_accelerator_base_module_get_handle_fn_t)(
+    void *dev_ptr, void **handle);
+
+/**
+ * Opens an interprocess memory handle from another process and returns
+ * a device pointer usable in the local process.
+ *
+ * @param[IN] handle         IPC memory handle from another process
+ * @param[IN] dev_id         Device ID associated with the IPC memory handle
+ * @param[OUT] dev_ptr       Returned device pointer
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_open_handle_fn_t)(
+    void **handle, int dev_id, void **dev_ptr);
+
+/**
+ * Closes memory mapped with opal_accelerator_base_module_open_handle_fn_t().
+ *
+ * @param[IN] dev_ptr            IPC device pointer returned from
+ *                               opal_accelerator_base_module_open_handle_fn_t()
+ */
+typedef int (*opal_accelerator_base_module_close_handle_fn_t)(
+    void *dev_ptr);
+
+/**
+ * Page-locks the memory range specified by ptr and size
+ *
+ * @param[IN] ptr            Host pointer to memory to page-lock
+ * @param[IN] size           Size in bytes of the address range to page-lock in bytes
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_host_register_fn_t)(
+    void *ptr, size_t size);
+
+/**
+ * Unregisters a memory range that was registered with
+ * opal_accelerator_base_module_host_register_fn_t.
+ *
+ * @param[IN] ptr            Host pointer to memory to unregister
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_host_unregister_fn_t)(
+    void *ptr);
+
+/**
+ * Retrieves current device id for a device associated with the local process.
+ *
+ * @param[OUT] dev_id        ID of the device
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_get_device_fn_t)(
```

Comment:
I thought you said that Ze didn't have this assumption and could use multiple devices in a single process?

Reply:
So you're maintaining the assumption that a single process "controls" at most one accelerator.  Does that match with what Intel and/or AMD want?

Reply:
Assuming this would be implemented using cudaGetDevice/hipGetDevice, in all reality it would return which ever device has been set by the user in the code, unless you want to cache internally the value. The AMD ROCr runtime library does not have the notion of a 'current' device, it is purely maintained on the hip layer.

Reply:
Did we ever sort this out?

Reply:
Added device ids back in after a conversation with Sean Hefty from Intel